src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,327 @@
+          }
+        }
+      }
+      // for every other case where feature_name isn't 'intensity', i.e. 'peak_apex_int'
+      else
+      {
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          intensities_vec.push_back(it->getMetaValue(feature_name));
+        }
+        std::vector<OpenMS::Peak2D::IntensityType>::iterator max_it = std::max_element(intensities_vec.begin(), intensities_vec.end());
+        double measured_feature_max = intensities_vec[std::distance(intensities_vec.begin(), max_it)];
+          
+        for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+        {
+          if (measured_feature_max != 0.0)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(feature_name) / measured_feature_max);
+          }
+        }
+      }
+    }
+    else if (mass_intensity_type == "norm_sum")
+    {
+      if (feature_name == "intensity")
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += it->getIntensity();
+        }
+        
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          normalized_feature.setMetaValue((it - measured_feature_subordinates.begin()), (it->getIntensity() / feature_peak_apex_intensity_sum));
+        }
+      }
+      // for every other case where feature_name isn't 'intensity', i.e. 'peak_apex_int'
+      else
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += (Peak2D::IntensityType)it->getMetaValue(feature_name);
+        }
+
+        for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+        {
+          if (feature_peak_apex_intensity_sum != 0.0)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(feature_name) / feature_peak_apex_intensity_sum);
+          }
+        }
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateMDVs(
+    const FeatureMap& measured_featureMap, FeatureMap& normalized_featureMap,
+    const String& mass_intensity_type, const String& feature_name)
+  {
```

Comment:
would it make sense to `.clear()` the output, before pushing more stuff into it?

Reply:
`normalized_featureMap` is supposed to be just declared by the user but I will clear it to make sure `normalized_feature` are pushed back into an empty vector

Reply:
good. Never trust the input!
And if you can do the work (= `.clear()`) in one place, prefer that over doing it in many places (whenever the method is called).
And if the vector is already empty (because it was created), clearing it is a NOOP, so win-win.